tests/cmds/test_devices.py

```diff
@@ -0,0 +1,555 @@
+    cli_state.sdk.devices.get_by_guid.return_value = empty_backupusage_response
+
+
+@pytest.fixture
+def get_all_devices_success(cli_state, devices_list_generator):
+    cli_state.sdk.devices.get_all.return_value = devices_list_generator
+
+
+@pytest.fixture
+def get_all_users_success(cli_state, users_list_generator):
+    cli_state.sdk.users.get_all.return_value = users_list_generator
+
+
+def test_deactivate_deactivates_device(
+    runner, cli_state, deactivate_device_success, get_device_by_guid_success
+):
+    runner.invoke(cli, ["devices", "deactivate", TEST_DEVICE_GUID], obj=cli_state)
+    cli_state.sdk.devices.deactivate.assert_called_once_with(TEST_DEVICE_ID)
+
+
+def test_deactivate_when_given_flag_updates_purge_date(
+    runner,
+    cli_state,
+    deactivate_device_success,
+    get_device_by_guid_success,
+    device_info_success,
+    archives_list_success,
+):
+    runner.invoke(
+        cli,
+        ["devices", "deactivate", TEST_DEVICE_GUID, "--purge-date", TEST_PURGE_DATE],
+        obj=cli_state,
+    )
+    cli_state.sdk.archive.update_cold_storage_purge_date.assert_called_once_with(
+        TEST_ARCHIVE_GUID, TEST_PURGE_DATE
+    )
+
+
+def test_deactivate_when_given_flag_changes_device_name(
+    runner,
+    cli_state,
+    deactivate_device_success,
+    get_device_by_guid_success,
+    device_info_success,
+    mock_device_settings,
+):
+    cli_state.sdk.devices.get_settings.return_value = mock_device_settings
+    runner.invoke(
+        cli,
+        ["devices", "deactivate", TEST_DEVICE_GUID, "--change-device-name"],
+        obj=cli_state,
+    )
+    assert (
+        mock_device_settings.name
+        == "deactivated_" + date.today().strftime("%Y-%m-%d") + "_testname"
+    )
+    cli_state.sdk.devices.update_settings.assert_called_once_with(mock_device_settings)
+
+
+def test_deactivate_does_not_change_device_name_when_not_given_flag(
+    runner,
+    cli_state,
+    deactivate_device_success,
+    device_info_success,
+    mock_device_settings,
+):
+    cli_state.sdk.devices.get_settings.return_value = mock_device_settings
+    runner.invoke(
+        cli, ["devices", "deactivate", TEST_DEVICE_ID], obj=cli_state,
+    )
+    assert mock_device_settings.name == "testname"
+    cli_state.sdk.devices.update_settings.assert_not_called()
+
+
+def test_deactivate_fails_if_device_does_not_exist(
+    runner, cli_state, deactivate_device_not_found_failure
+):
+    result = runner.invoke(
+        cli, ["devices", "deactivate", TEST_DEVICE_ID], obj=cli_state
+    )
+    assert result.exit_code == 1
+    assert "The device {} was not found.".format(TEST_DEVICE_ID)
+
+
+def test_deactivate_fails_if_device_is_on_legal_hold(
+    runner, cli_state, deactivate_device_in_legal_hold_failure
+):
+    result = runner.invoke(
+        cli, ["devices", "deactivate", TEST_DEVICE_ID], obj=cli_state
+    )
+    assert result.exit_code == 1
+    assert "The device {} is in legal hold.".format(TEST_DEVICE_ID)
```

Comment:
all of these that are checking the output text need to assert that the text is `in result.ouput`, currently, they're just asserting the string, which will always be true since it's not empty.

Reply:
I guess there's just the 3 missing the `in result.output`

test_deactivate_fails_if_device_does_not_exist
test_deactivate_fails_if_device_is_on_legal_hold
test_deactivate_fails_if_device_deactivation_forbidden

Reply:
updated